app/models/apple/episode.rb

```diff
@@ -2,14 +2,59 @@
 
 module Apple
   class Episode
-    attr_accessor :show, :feeder_episode, :api
+    include Apple::ApiWaiting
+    include Apple::ApiResponse
+    attr_accessor :show, :feeder_episode, :api, :api_response
+
+    AUDIO_ASSET_FAILURE = "FAILURE"
+    AUDIO_ASSET_SUCCESS = "SUCCESS"
+
+    def self.wait_for_asset_state(api, eps)
+      wait_for(eps) do |remaining_eps|
+        Rails.logger.info("Probing for episode audio asset state")
+        unwrapped = get_episodes(api, remaining_eps)
+
+        remote_ep_by_id = unwrapped.map { |row| [row["request_metadata"]["guid"], row] }.to_h
+        remaining_eps.each { |ep| ep.api_response = remote_ep_by_id[ep.guid] }
+
+        rem =
+          remaining_eps.filter do |ep|
+            if ep.waiting_for_asset_state?
+              Rails.logger.info("Waiting for audio asset state", {episode_id: ep.feeder_id,
+                                                                      delivery_file_count: ep.podcast_delivery_files.count,
+                                                                      delivery_files_processed_errors: ep.podcast_delivery_files.all?(&:processed_errors?),
+                                                                      delivery_files_processed: ep.podcast_delivery_files.all?(&:processed?),
+                                                                      delivery_files_delivered: ep.podcast_delivery_files.all?(&:delivered?),
+                                                                      asset_state: ep.audio_asset_state,
+                                                                      has_podcast_audio: ep&.podcast_container&.has_podcast_audio?})
+              true
+            end
+          end
+
+        if rem.length > 0
+          Rails.logger.info("Waiting for asset state processing", {audio_asset_states: rem.map(&:audio_asset_state).uniq})
+        end
+
+        rem
+      end
+    end
 
     def self.get_episodes(api, episodes)
       return if episodes.empty?
 
       api.bridge_remote_and_retry!("getEpisodes", episodes.map(&:get_episode_bridge_params))
     end
 
+    def self.get_episodes_via_show(api, show_id)
+      bridge_params = Apple::Show.get_episodes_json(api, show_id).map do |ep_json|
+        id = ep_json["id"]
+        guid = ep_json["attributes"]["guid"]
+        Episode.get_episode_bridge_params(api, id, guid)
+      end
+
+      api.bridge_remote_and_retry!("getEpisodes", bridge_params)
+    end
+
```

Comment:
Now the episodes have the same pattern as the PodcastContainer/PodcastDelivery/PodcastDeliveryFile where you can see the pattern as `.get_<children>_via_<parent>`.

1) Use the 'parent' model as a means to get the index view (e.g. `show/<show_id>/relationships/episodes`)
2) Then use singular api endpoint (e.g. `episodes/<ep_id>`) to get a standardized view of the resource data